data-generation/src/main/scala/thesis/Landy.scala

```diff
@@ -72,19 +72,38 @@ class Landy(graph: LandyAttributeGraph) extends TemporalGraph[LandyEntityPayload
     this.vertices.filter(vertex => vertex._2 != null)
   }
 
-  /** Return entity based on ID at a timestamp
-   *
-   * @param entity    Edge or Vertex
-   * @param timestamp Insant
-   * @return Entity Object and HashMap
-   */
-  override def getEntity[T <: Entity](entity: T, timestamp: Instant): Option[(T, Attributes)] = ???
+  override def getEntity[T <: Entity](entity: T, timestamp: Instant): Option[(T, Attributes)] = {
+    entity match {
+      case _: VERTEX => getVertex(entity, timestamp)
+      case _: EDGE => getEdge(entity, timestamp)
```

Comment:
Vil anta du har hatt noe typeproblemer. Når man pattern matcher skjønner kompilatoren hvilken konkret type den er. 
```suggestion
      case v: VERTEX => getVertex(v, timestamp)
      case e: EDGE => getEdge(e, timestamp)
```

Reply:
Henger ikke helt med. Hvilke fordeler får man ved å mellomlagre den til `e` / `v` og så sende den inn i funksjonen? 🤔